pandas/io/formats/format.py

```diff
@@ -1237,6 +1237,8 @@ def _format(x):
                 fmt_values.append(f" {_format(v)}")
             elif is_float_type[i]:
                 fmt_values.append(float_format(v))
+            elif isinstance(v, str):
```

Comment:
GenericArrayFormatter has a `quoting` parameter. format_array does not have this parameter to be able to pass it on. 

`_repr_categories` in `pandas\core\arrays\categorical.py` uses `format_array`.

is it feasible to add `quoting` parameter to format_array instead of changing this?

Reply:
@simonjayhawkins thanks for the suggestion!

So, in `_repr_categories` from `pandas\core\arrays\categorical.py`, `format_array` is called. This could be changed to, say,
```python
category_strs = fmt.format_array(self.categories, None, quoting=2)
```
(quoting=2 quotes non-numeric, as far as I understand), and then, inside `format_array`, pass `quoting` on to `fmt_klass`, so that (here) `GenericArrayFormatter` would be initialised with it.

It's unclear to me me what `GenericArrayFormatter` currently does with `quoting`, I don't see it being used. Should there be no need to change anything here in `_format_strings`?

Reply:
> It's unclear to me me what `GenericArrayFormatter` currently does with `quoting`, I don't see it being used. Should there be no need to change anything here in `_format_strings`?

that does seem strange, I must have assumed wrongly where the quoting parameter is used. will look again, but your suggestions sgtm.

Reply:
instead of this you need to adjust _format, L1221; you maybe able to simply adjust the args to pprint_thing which can quote as needed.

Reply:
Yes, thanks for pointing me towards that, that was useful!

So, `pprint_thing` takes a `quote_string` argument, which is boolean, while `quoting` is `Optional[int]`, so I've tried doing this by setting
```python
quote_strings = self.quoting is not None and self.quoting != QUOTE_NONE
```
and then passing `quote_strings` to `pprint_thing`